sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/ComplexTypes.scala

```diff
@@ -59,6 +63,13 @@ object SimplifyExtractValueOps extends Rule[LogicalPlan] {
           Literal(null, ga.dataType)
         }
       case GetMapValue(CreateMap(elems), key) => CaseKeyWhen(key, elems)
+      case GetMapValue(Literal(map: MapData, MapType(kt, vt, _)), key) =>

```

Comment:
Since the constant folding covers foldable cases, we need to check if key is unfoldable?

Reply:
I don't think so. Constant folding is executed before this, hence if the key is foldable we never get here. So the check is useless. We can add it and a comment for explaining when this case happens if you think is needed, but it'd be anyway quite clear from the history and the description of this PR...

Reply:
Since we can remove the rule thru `spark.sql.optimizer.excludedRules`, I feel we'd better to add the check. I leave it to other reviewers.

Reply:
+1 for @maropu 's comment. Let's add `if` check or `assert` or the comment about the assumption.